### PR TITLE
[WIP] Two GPUs + CUDF

### PR DESCRIPTION
**Pull request by quasiben**
cc @mrocklin @Akshay-Venkatesh 

This PR adds a test for starting "workers" attached to separate GPUs.  This is intended to mimic a failing/halting case which we regularly see when using a similar message passing scheme with dask.

Tests currently pass for the following cases:

- cdf = cudf.DataFrame({"a": range(5), "b": range(
- cdf = cudf.DataFrame({"a": [1.0], "b": [1.0]}).head(0)
 
This is somewhat disappointing -- I was hoping at least one of the above would fail and help narrow down the issue

### CODE REVIEW

File: tests/test_send_recv_two_workers.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            frames.append(frame)
+
+        header = pickle.loads(frames[0])
+        frames = frames[1:]  #
+        typ = pickle.loads(header["type"])
+        cuda_obj = typ.deserialize(header, frames)
+
+        print(cuda_obj, len(cuda_obj))
+        asyncio.sleep(5)
+        # how should we confirm data is on two GPUs ?
```
Comment (quasiben) [inline tests/test_send_recv_two_workers.py]: @kkraus14 / @pentschev  at this point in the test we have sent data from GPU1 to GPU5.  Do you know of an easy way to confirm data on both GPUs ?

**Comment by quasiben** [comment]
One nice thing about this style of test is that it is self contained -- no env vars need to be defined:

> py.test -s -v tests/test_send_recv_two_workers.py

File: tests/test_send_recv_two_workers.py (hunk 1 above)
Comment (mrocklin) [reply tests/test_send_recv_two_workers.py]: You might want to refer to this pynvml code that we use within distributed to get the memory use and utilization of each GPU: https://github.com/dask/distributed/blob/master/distributed/diagnostics/nvml.py

File: tests/test_send_recv_two_workers.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+            data = cupy.arange(10)
+            frames = await distributed.comm.utils.to_frames(
+                data, serializers=("cuda", "pickle")
+            )
+        if cuda_type == "cudf":
+            import cudf
+            import numpy as np
+            asyncio.sleep(0.01)
+            # cdf = cudf.DataFrame({"a": range(5), "b": range(5)})
```
Comment (quasiben) [inline tests/test_send_recv_two_workers.py]: The cudf DFs should be parameterized as part of the test

File: tests/test_send_recv_two_workers.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        print("Last Bit of Cleanup...")
+        ucp.fin()
+
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(f(port))
+
+
+def test_send_recv_cudf(event_loop):
+    base_env = {
+        "UCX_RNDV_SCHEME": "put_zcopy",
```
Comment (quasiben) [inline tests/test_send_recv_two_workers.py]: We should parameterize the CUCDA_VISIBLE DEVICE for different configurations and we should also test when workers are attached to the same ib interface and difference interfaces: (ib0, ib1)

File: tests/test_send_recv_two_workers.py (hunk 1 above)
Comment (kkraus14) [reply tests/test_send_recv_two_workers.py]: You can also use `gdf['a'].data.mem.gpu_data.context.PCI_BUS_ID` which will return the unique PCI bus identifier of the GPU (will not be 0 - 7 on a DGX for example)

**Pull request closed without merging**